File: linguagem_java/encapsulamento/empresa/Funcionario.java
package encapsulamento.empresa;

public class Funcionario {
    // Atributos da classe Funcionario
    private String nomeDoFuncionario;
    private Empresa empresa;

    // Construtor da classe Funcionario
    Funcionario(String nomeDoFuncionario, Empresa empresa) {
        setNomeDoFuncionario(nomeDoFuncionario);
        setEmpresa(empresa);
    }

    // Getter e setter de empresa
    public Empresa getEmpresa() {
        return empresa;
    }

    public void setEmpresa(Empresa empresa) {
        this.empresa = empresa;
    }

    // Getter e setter de nomeDoFuncionario
    public String getNomeDoFuncionario() {
        return nomeDoFuncionario;
    }

    public void setNomeDoFuncionario(String nomeDoFuncionario) {
        this.nomeDoFuncionario = nomeDoFuncionario;
    }

    // Método que retorna informações
    String informarSobreFuncionario() {
        return "\nO funcionário " + nomeDoFuncionario + " trabalha na empresa " + empresa.getNomeDaEmpresa();
    }

}

File: linguagem_dart/encapsulamento/empresa/funcionario.dart
import 'empresa.dart';

class Funcionario {
  // Atributos da classe Funcionario
  String _nomeDoFuncionario = "";
  late Empresa _empresa;

  // Construtor da classe Funcionario
  Funcionario(String nome, Empresa empresa) {
    setNomeDoFuncionario = nome;
    setEmpresa = empresa;
  }

  // Getter e setter de Empresa
  Empresa get getEmpresa => _empresa;

  set setEmpresa(Empresa empresa) {
    _empresa = empresa;
  }

  // Getter e setter de nomeDoFuncionario
  String get getNomeDoFuncionario => _nomeDoFuncionario;

  set setNomeDoFuncionario(String nomeDoFuncionario) {
    _nomeDoFuncionario = nomeDoFuncionario;
  }

  // Método que retorna informações
  String retornarInformacoesFuncionario() {
    return "\nO funcionário $_nomeDoFuncionario trabalha na empresa ${_empresa.getNomeDaEmpresa}\n";
  }
}
